src/transformers.jl

```diff
@@ -72,7 +72,11 @@ function apply(A::AbstractArray, t::Transform; dims=:, inds=:, kwargs...)
         end
     end
 
-    return mapslices(x -> _apply(x[inds], t; kwargs...), A, dims=dims)
+    slice_index = 0
+    return mapslices(A, dims=dims) do x
+        slice_index += 1
```

Comment:
I couldn't find a way to enumerate directly. [Suggestions from discourse](https://discourse.julialang.org/t/is-there-a-way-to-enumerate-mapslices-other-than-i-0-result-mapslices-m/55414) use other packages/functions.

Reply:
Does mapslices support multiple dims like dims=[1,2] and if it does, does that cause issues with slice indexing like this? Also if, dims=2 (not the default), does the corresponding std and mean named tuples know that it doesn't apply to the columns anymore?

Reply:
> Does mapslices support multiple dims like dims=[1,2] and if it does, does that cause issues with slice indexing like this? 

It does support multiple dims like that. I expect it would cause an issue somewhere, just because we haven't tested it and Murphy's Law. But I don't see what the problem would be - as long as `dims` is consistent between constructing `MeanStdScaling` and applying, the `slice_index` will correspond. `slice_index` is just numbering the slices, whatever those slices are.

> Also if, dims=2 (not the default), does the corresponding std and mean named tuples know that it doesn't apply to the columns anymore?

Not sure I understand right, but again as long as `dims` is the same between constructing and applying, the named tuples will be right. I was thinking about adding `dims` as a field, to check this / avoid specifying `dims` twice, but I want to leave that for now.

Reply:
Ok I think this is fine then